Makefile

```diff
@@ -0,0 +1,191 @@
+# executables
+docker := $(shell which docker) --log-level error
+gcloud := $(shell which gcloud) --quiet --verbosity=error
+git := $(shell which git)
+go := $(shell which go)
+kubectl := $(shell which kubectl)
+
+# plugin's binary options
+binary_name = drone-gke
+
+# git options
+git_default_branch = $(shell cat $(CURDIR)/.git/refs/remotes/origin/HEAD | cut -d ' ' -f 2 | cut -d '/' -f 4)
+git_current_branch = $(shell $(git) rev-parse --abbrev-ref HEAD)
+git_current_revision = $(shell $(git) rev-parse --short HEAD)
+
+# plugin's docker options
+docker_default_repo_name = nytimes
+docker_image_name = drone-gke
+docker_default_tag = latest
+ifneq ($(git_current_branch), $(git_default_branch))
+	docker_default_tag = $(git_current_branch)
+endif
+docker_default_file = $(CURDIR)/Dockerfile
+docker_default_build_context = $(CURDIR)
+
+# gcloud SDK options
+gcloud_sdk_image_name = google/cloud-sdk
+gcloud_default_sdk_version = alpine
+
+# print usage (default target)
+usage :
+	@echo "Usage:"
+	@echo ""
+	@echo "  \$$ make check               # run tests for required software"
```

Comment:
the commands would fail, but the thinking behind these `check-*` targets is that each could be added as a prerequisite to another target .. effectively becoming a hook to provide some feedback to a user about installing whatever dependency they're missing:

```makefile
# compile binary
$(binary_name) : check-go # ...
```

if the user doesn't have go installed, we could include something like a [install documentation](https://golang.org/doc/install)

that said, not at all opposed to removing these.. is that what you'd prefer?

Reply:
Since it's written, let's keep it.
I think if it becomes a cause of errors then we can remove it then to reduce points of maintenance :)